### PR TITLE
Improve GNMI incremental config and add unit test for BGP

### PR DESCRIPTION
**Pull request by ganglyu**
#### Why I did it
GCU has verified BGP configuration, we need to verify that GNMI can support the same BGP configuration.
Because GCU end2end test uses jsonpatch replace operation for BGP, we need to update GNMI replace operation for incremental config.
Golang does not have reliable library for jsonpatch.
Microsoft ADO: 27231737

#### How I did it
This unit test uses BGP configuration from GCU test.
This unit test generates GNMI request for BGP and use jsonpatch to verify patch file generated by GNMI server.
I also update gnmi replace implementation to support jsonpatch replace operation.
"config apply-patch" is equivalent to “config replace", so I use "config replace" for incremental config, and then we don't need to generate jsonpatch in golang.

#### How to verify it
Run gnmi unit test.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: test/test_gnmi_configdb.py [hunk 1]
```diff
@@ -252,7 +252,7 @@ def test_gnmi_incremental_replace(self, test_data):
             test_path = item['path']
             test_value = item['value']
             for patch_data in patch_json:
-                assert patch_data['op'] == 'add', "Invalid operation"
```
Comment (qiluo-msft) [inline test/test_gnmi_configdb.py:255]: >add [](http://example.com/codeflow?start=44&length=3)

you should also keep old testcase if it still make sense

File: sonic_data_client/mixed_db_client.go [hunk 2]
```diff
@@ -1110,7 +1113,7 @@ func (c *MixedDbClient) handleTableData(tblPaths []tablePath) error {
 }
 
 /* Populate the JsonPatch corresponding each GNMI operation. */
-func (c *MixedDbClient) ConvertToJsonPatch(prefix *gnmipb.Path, path *gnmipb.Path, t *gnmipb.TypedValue, output *string) error {
+func (c *MixedDbClient) ConvertToJsonPatch(prefix *gnmipb.Path, path *gnmipb.Path, t *gnmipb.TypedValue, operation string, output *string) error {
```
Comment (qiluo-msft) [inline sonic_data_client/mixed_db_client.go:1116]: >operation [](http://example.com/codeflow?start=105&length=9)

Could you PR title or PR description for this new feature?

Comment (ganglyu) [reply sonic_data_client/mixed_db_client.go:1116]: updated.

File: sonic_data_client/mixed_db_client.go [hunk 3]
```diff
@@ -1122,11 +1125,7 @@ func (c *MixedDbClient) ConvertToJsonPatch(prefix *gnmipb.Path, path *gnmipb.Pat
 	}
 
 	elems := fullPath.GetElem()
-	if t == nil {
-		*output = `{"op": "remove", "path": "/`
-	} else {
-		*output = `{"op": "add", "path": "/`
-	}
+	*output = `{"op": "` + operation + `", "path": "/`
```
Comment (qiluo-msft) [inline sonic_data_client/mixed_db_client.go:1128]: >op [](http://example.com/codeflow?start=14&length=2)

Manual serialization is dangerous. Will you just construct the json object and use library to serialize it.

**Pull request closed without merging**